src/app/Scenes/Activity/components/ExpiresInTimer.tsx

```diff
@@ -0,0 +1,84 @@
+import { Stopwatch, Flex, Text } from "@artsy/palette-mobile"
+import { ActivityItem_item$data } from "__generated__/ActivityItem_item.graphql"
+import { Time, getTimer } from "app/utils/getTimer"
+import { useEffect, useRef, useState } from "react"
+
+const INTERVAL = 1000
+
+export const ExpiresInTimer: React.FC<{ expiresAt: string }> = ({ expiresAt }) => {
+  const intervalId = useRef<ReturnType<typeof setInterval> | null>(null)
+
+  const [time, setTime] = useState<Time>(getTimer(expiresAt)["time"])
+  const [hasEnded, setHasEnded] = useState(getTimer(expiresAt)["hasEnded"])
+
+  useEffect(() => {
+    intervalId.current = setInterval(() => {
+      const { hasEnded: timerHasEnded, time: timerTime } = getTimer(expiresAt)
+
+      setHasEnded(timerHasEnded)
+      setTime(timerTime)
+    }, INTERVAL)
+
+    return () => {
+      // @ts-expect-error PLEASE_FIX_ME_STRICT_NULL_CHECK_MIGRATION
```

Comment:
Maybe better than a @ts-expect-error 🤔 
```suggestion
      if(!intervalId.current) return
```

Reply:
thank, fixed!